### DISCUSSION TITLE
MAUI Windows Production Sample

### DISCUSSION THREAD

**Discussion by daltzctr** [dotnet/maui#14897] [General]
### Description

As an individual looking for a cross-platform technology that primarily targets Windows, with Android/iOS secondary targets, I'm investigating MAUI as a potential options. However, looking at the official samples located at https://github.com/dotnet/maui-samples/tree/main/7.0/UserInterface/Views and extra samples available at 
https://github.com/jsuarezruiz/awesome-dotnet-maui, I am wary about it's readiness state for a production project. 

In the above linked examples, nearly all of them are Android/iOS **only** with the few that explicitly support Windows obviously having several issues when actually running them. For example, in the Weather 21 sample, I've observed 

- Slow navigation
- Broken color schemes
- Poor responsiveness on resizing

And wondering if this is what I should expect when developing my own application, when the official sample itself is poor? 

In the instance that the sample itself is just outdated, can I get pointed to a production quality sample that utilizes MAUI (non-Blazor) for a Windows desktop application. Something that is actually Windows tested, and not just "oh yeah it works on Windows"?

I do not intend for this issue to be antagonistic but am merely looking for straightforward answers. Upon talking with multiple individuals on both the .NET development and Help discord, and research on various forums, it seems like nobody has ever gotten a direct response.

Thanks!

### Public API Changes

More Windows Samples

### Intended Use-Case

Better focus on Windows development

**Comment by PureWeen** [comment]
@davidortinau @davidbritch

**Reply by Xyncgas** [reply]
Since Xamarin wasn't running on window, MAUI can, it's exciting to see toast notification & modifying application window & etc

**Reply by daltzctr** [reply]
Xamarin runs on Windows, but it's still not a good experience.

**Comment by yyliveyy** [comment]
Are you counting on a framework without a datagrid to develop a desktop? Wake up, MAUI was not developed for desktop applications from the beginning.

**Reply by daltzctr** [reply]
So

> Using .NET MAUI, you can develop apps that can run on Android, iOS, macOS, and Windows from a single shared code-base.

Should become

> Using .NET MAUI, you can develop apps that can run on Android and iOS

**Comment by Xyncgas** [comment]
Would be amazing if we can get React native experience on MAUI.Blazor, instead of interacting with the file system in .NET code by making UI talk with .NET, I might as well use react components and its browser native experience to do it, while I do some other stuffs with .NET

**Reply by Xyncgas** [reply]
It still feels better to write layout algorithms in F# than in javascript

**Comment by heron1** [comment]
I made the silly mistake of writing an app I intended for production without thinking about deployment properly, and went off the marketing hype alone. From my experience the app cannot be deployed reliably as a standalone executable without purchasing yearly keys since it deploys as a .msix package only. The team also seem to have not given any indication if native .exe support will ever be available.

**Reply by Xyncgas** [reply]
Since MAUI is open source perhaps you can compile it to exe once you glue it with other UI frameworks

**Reply by heron1** [reply]
yes or even just use webview in winforms. But I fell for the marketing hype that something new and shiny was going to just make a single framework that did everything, when in reality MAUI is a mobile framework it seems.

**Reply by Xyncgas** [reply]
it's almost if microsoft 3D is being compared with blender
